File: 백준/Silver/6159. 코스튬 파티/코스튬 파티.cc
#include "iostream"
#include "vector"

using namespace std;

int costumeParty(int s, vector<int> cows) {
    int cnt = 0;
    for (int i = 0; i < cows.size(); ++i) {
        for (int j = i + 1; j < cows.size(); ++j) {
            if (cows[i] + cows[j] <= s) {
                cnt++;
            }
        }
    }
    return cnt;
}


int main() {
    int n, s;
    scanf("%d %d", &n, &s);

    vector<int> cows;
    for (int i = 0; i < n; ++i) {
        int size;
        scanf("%d", &size);
        cows.push_back(size);
    }

    int cnt = costumeParty(s, cows);
    printf("%d \n", cnt);
}

File: 백준/Silver/6159. 코스튬 파티/코스튬 파티.py
import sys


def costume_party(cows):
    count = 0
    for i in range(len(cows)):
        for j in range(i + 1, len(cows)):
            sum_size = cows[i] + cows[j]
            if sum_size <= s:
                count += 1

    return count


n, s = map(int, sys.stdin.readline().rstrip().split())
arr = [int(sys.stdin.readline().rstrip()) for _ in range(n)]
print(costume_party(arr))
